File: 06/Java/TripleExpression/Add.java
package TripleExpression;

public class Add extends BinaryOperator implements TripleExpression {
    public Add(TripleExpression first, TripleExpression second) {
        super(first, second);
    }

    protected int apply(int a, int b) {
        return a + b;
    }
}

File: 08/CSharp/TripleExpressionApp/TripleExpressionApp/Add.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Add<T> : BinaryOperator<T>
    {
        public Add(ITripleExpression<T> first, ITripleExpression<T> second, IOperator<T> op)
        {
            fOp = first;
            sOp = second;
            base.op = op;
        }
        
        protected override T Apply(T a, T b)
        {
            return op.Add(a, b);
        }
    }
}
